File: src/Training.FileExplorer.ServerApp/Training.FileExplorer.Application/FileStorage/Models/Filtering/StorageDriveEntryFilterModel.cs
using Training.FileExplorer.Application.Common.Models.Filtering;

namespace Training.FileExplorer.Application.FileStorage.Models.Filtering;

public class StorageDriveEntryFilterModel : FilterPagination
{
    public string DriveLabel { get; set; } = string.Empty;

    public bool IncludeFiles { get; set; }

    public bool IncludeDirectories { get; set; }
}

File: src/Training.FileExplorer.ClientApp/src/infrastructure/models/filtering/StorageDriveEntryFilterModel.ts
import { FilterPagination } from "@/infrastructure/models/filtering/FilterPagination";

export class StorageDriveEntryFilterModel extends FilterPagination {
    includeDirectories: boolean;
    includeFiles: boolean;

    constructor(pageSize: number, pageToken: number, includeDirectories: boolean, includeFiles: boolean) {
        super(pageSize, pageToken);

        this.includeDirectories = includeDirectories;
        this.includeFiles = includeFiles;
    }

    override convertToQueryParams(): URLSearchParams {
        const params = super.convertToQueryParams();
        params.append("includeDirectories", this.includeDirectories.toString());
        params.append("includeFiles", this.includeFiles.toString());

        return params;
    }
}